internal/services/webhooks/webhooks.go

```diff
@@ -0,0 +1,317 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/hatchet-dev/hatchet/internal/whrequest"
+	"github.com/hatchet-dev/hatchet/pkg/config/server"
+	"github.com/hatchet-dev/hatchet/pkg/repository"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/dbsqlc"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/sqlchelpers"
+	"github.com/hatchet-dev/hatchet/pkg/webhook"
+)
+
+type WebhooksController struct {
+	sc                  *server.ServerConfig
+	registeredWorkerIds map[string]bool
+	cleanups            []func() error
+}
+
+func New(sc *server.ServerConfig) *WebhooksController {
+	return &WebhooksController{
+		sc:                  sc,
+		registeredWorkerIds: map[string]bool{},
+	}
+}
+
+func (c *WebhooksController) Start() (func() error, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	ticker := time.NewTicker(30 * time.Second)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				if err := c.check(); err != nil {
+					c.sc.Logger.Warn().Err(err).Msgf("error checking webhooks")
+				}
+			case <-ctx.Done():
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+
+	return func() error {
+		cancel()
+
+		for _, cleanup := range c.cleanups {
+			if err := cleanup(); err != nil {
+				return fmt.Errorf("could not cleanup webhook worker: %w", err)
+			}
+		}
+
+		return nil
+	}, nil
+}
+
+func (c *WebhooksController) check() error {
+	tenants, err := c.sc.EngineRepository.Tenant().ListTenants(context.Background())
+	if err != nil {
+		return fmt.Errorf("could not list tenants: %w", err)
+	}
+
+	for _, tenant := range tenants {
+		tenantId := sqlchelpers.UUIDToStr(tenant.ID)
+
+		wws, err := c.sc.EngineRepository.WebhookWorker().ListWebhookWorkers(context.Background(), tenantId)
+		if err != nil {
+			return fmt.Errorf("could not get webhook workers: %w", err)
+		}
+
+		for _, ww := range wws {
+			id := sqlchelpers.UUIDToStr(ww.ID)
+			if _, ok := c.registeredWorkerIds[id]; ok {
+				if ww.Deleted {
+					c.sc.Logger.Debug().Msgf("webhook worker %s of tenant %s has been deleted", id, tenantId)
+					err := c.sc.EngineRepository.Worker().DeleteWorkersByName(context.Background(), tenantId, "Webhook_"+id)
```

Comment:
Why aren't we just using the `DeleteWorker` method for this? Don't we have access to the associated worker id?

Reply:
we don't have access to the worker ID, no. We would need to track it in the DB and while using the worker name does not seem perfect, neither seems tracking this additionaly in the DB because you mentioned that you do not want too much webhook workers logic as this ideally could be done outside of the engine as well

Reply:
Ok, that makes sense - but does it even make sense to delete the worker vs mark the worker as inactive? We don't currently delete workers, because it's useful to look back at the run history sometimes.